Deploy-CAM.ps1

```diff
@@ -3170,32 +3258,192 @@ if(-not $keyVaultProviderExists) {
     Register-AzureRmResourceProvider -ProviderNamespace "Microsoft.KeyVault" -ErrorAction stop | Out-Null
 }
 
-# Find the CAM root RG.
-$resouceGroups = Get-AzureRmResourceGroup
+# Determine if we're upgrading a current deployment
+# This section returns a resource group name in $ResourceGroupName if one was found.
+$CAMKeyVaults = Find-AzureRmResource -ResourceType "Microsoft.KeyVault/vaults" | Where-object {$_.Name -like "CAM-*"}
+
+$deploymentIndex = 0
+ForEach ($s in $CAMKeyVaults) {
+    if (-not (Get-Member -inputobject $s -name "Number")) {
+        Add-Member -InputObject $s -Name "Number" -Value "" -MemberType NoteProperty
+    }
+
+    $s.Number = ++$deploymentIndex
+}
+
+if((-not $ResourceGroupName) -and ($deploymentIndex -gt 0)) {
+    while ($true) {
+
+        Write-Host "`nCloud Access Manager deployments in this subscription"
+        Write-Host ($CAMKeyVaults | Select-Object -Property Number, ResourceGroupName | Format-Table | Out-String)
+    
+    
+        Write-Host ("To upgrade or modify a Cloud Access Manager Deployment, select the resource group of the`n" +
+           "Cloud Access Manager deployment root by number, or hit enter to create a new deployment.")
+        $rgIdentifier = (Read-Host "Resource group").Trim()
+        
+        # If empty string just exit loop
+        if(-not $rgIdentifier) {
+            break
+        }
+    
+        $rgIndex = 0
+        $rgIsInt = [int]::TryParse($rgIdentifier, [ref]$rgIndex) # rgIndex will be 0 on parse failure
+    
+        $rgArrayLength = $CAMKeyVaults.Length
+        if ( -not (( $rgIndex -ge 1) -and ( $rgIndex -le $rgArrayLength))) {
+            # Invalid range try again
+            Write-Host-Warning "Please enter a range between 1 and $rgArrayLength, or hit enter to create a new deployment"
+        }
+        else {
+            # Use the selected resource group for selection below
+            $ResourceGroupName = $CAMKeyVaults[$rgIndex - 1].ResourceGroupName
+            break
+        }
+    }
+}
+
+# Now we (might) have a $ResourceGroupName that (might) point to a resource group that has a CAM keyvault in it.
+# Let's double check what type.
+# At this point we have a subscription and a root resource group - check if there is already a deployment in it
+try{
+    $CAMRootKeyvault = $null
```

Comment:
Not a big deal, but instatiation should probably be outside the try, catch

Reply:
Good point. Fixed.